pkg/api/validate/dynamic/dynamic.go

```diff
@@ -287,11 +287,24 @@ func (dv *dynamic) validateCIDRRanges(ctx context.Context, subnets []Subnet, add
 			return err
 		}
 
-		_, net, err := net.ParseCIDR(*s.AddressPrefix)
-		if err != nil {
-			return err
+		// Since AddressPrefix seems to be deprecated for addressPrefixes we need this to validate the CIDR of the new
+		// field AddressPrefixes
+		// TODO: In theory we should be able to only check AddressPrefixes in the future
```

Comment:
Can we find out now? So we don't have to have this to do?

Reply:
@m1kola I was told by @bennerv not to make that change yet either way because of dependencies? @bennerv could you give some more insight?

Reply:
@m1kola @ArielLima 

`addressPrefixes` was moved from its own `AddressSpace` property to a top-level array property in the 2018-08-01 Network Resource provider release.  I don't think they're deprecating `addressPrefix` (singular) so we have to account for both. 

If you create a vnet / subnet address space using the current azure-cli, it will populate `addressPrefixes` (plural), but `addressPrefix` (singular) will be null.  

The `addressPrefixes` (plural) is used for dual-stack (ipv4/ipv6) networks, but the two are not mutually exclusive (you can have `addressPrefixes` specified without any ipv6 address space).  

I think this solution is the right way forward although it seems like a hack, but we have to account for the design of the NRP's API when we enumerate these subnets.

Reply:
@ArielLima 
It's possible they'll deprecate the field in the future, but we will hit that problem when we bump the consumed API version in the azure-sdk-for-go.  For now it might make sense to just remove the comment about deprecation and explain that you can specify it in multiple places and both are valid configs according to the NRP.